### PR TITLE
PI-1678: remove old methods

### PR DESCRIPTION
**Pull request by harry-brown-cko**
## Proposed changes

- Remove all unused public structs and methods
- Use Checkout SDK data types when available, deleting Frames versions

## Types of changes

What types of changes does your code introduce to Frames Ios?
_Put an `x` in the boxes that apply_

* [ ] Bugfix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

* [X] Lint and unit tests pass locally with my changes
* [X] I have added tests that prove my fix is effective or that my feature works
* [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: Source/CardUtils.swift [hunk 1]
```diff
@@ -1,65 +1,19 @@
...
-                           name: "Maestro",
-                           // swiftlint: disable line_length
-                           pattern: "^(?:5[06789]\\d\\d|(?!6011[0234])(?!60117[4789])(?!60118[6789])(?!60119)(?!64[456789])(?!65)6\\d{3})\\d{8,15}$",
-                           // swiftlint: enable line_length_violation
-                           gaps: [4, 8, 12],
-                           validLengths: [12, 13, 14, 15, 16, 17, 18, 19],
-                           validCvvLengths: [3],
-                           isLuhnChecked: true)
-    ]
+    let cardGaps: [Card.Scheme: [Int]] = [.visa: [4, 8, 12],
```
Comment (deepesh-vasthimal-cko) [inline Source/CardUtils.swift:9]: Is this part of checkout iOS SDK ?

Comment (harry-brown-cko) [reply Source/CardUtils.swift:9]: No

File: Source/CardUtils.swift [hunk 2]
```diff
@@ -117,90 +32,24 @@ public class CardUtils {
...
-        var numberOfGaps = 0
-        return cardNumber.reduce("") {
-            defer { index += 1 }
-            guard cardType.gaps.count > numberOfGaps else { return "\($0)\($1)" }
-            if index == cardType.gaps[numberOfGaps] {
-                numberOfGaps += 1
-                return "\($0) \($1)"
-            }
-            return "\($0)\($1)"
+    public func format(cardNumber: String, scheme: Card.Scheme) -> String {
```
[the remark this answers is not in the document]
Comment (harry-brown-cko) [reply Source/CardUtils.swift:35]: No

File: Source/CheckoutSdkIos.swift [hunk 3]
```diff
@@ -4,7 +4,7 @@ import UIKit
 /// Add a keyboard toolbar with navigation buttons. Use it to navigate between text fields.
 ///
 /// - parameter textFields: Array of text fields
-public func addKeyboardToolbarNavigation(textFields: [UITextField]) {
```
Comment (deepesh-vasthimal-cko) [inline Source/CheckoutSdkIos.swift:7]: Whats with the file name Source/CheckoutSdkIos.swift ?

File: Source/CheckoutSdkIos.swift [hunk 4]
```diff
@@ -4,7 +4,7 @@ import UIKit
...
+func addKeyboardToolbarNavigation(textFields: [UITextField]) {
```
Comment (deepesh-vasthimal-cko) [inline Source/CheckoutSdkIos.swift:7]: Question : what is the purpose of addKeyboardToolbarNavigation

File: Source/UI/CardListCell.swift [hunk 5]
```diff
@@ -76,10 +77,36 @@ public class CardListCell: UITableViewCell {
 
     /// Set the scheme icon of the card cell.
     ///
-    /// - parameter scheme: Scheme (e.g. CardScheme.visa)
-    public func setSchemeIcon(scheme: CardScheme) {
-        let image = "schemes/icon-\(scheme.rawValue)".image(forClass: CardListCell.self)
+    /// - parameter scheme: Scheme (e.g. Card.Scheme.visa)
+    public func setSchemeIcon(scheme: Card.Scheme) {
```
Comment (deepesh-vasthimal-cko) [inline Source/UI/CardListCell.swift:81]: Is there unit test for setSchemeIcon ? already

File: Source/UI/CardListCell.swift [hunk 6]
```diff
@@ -76,10 +77,36 @@ public class CardListCell: UITableViewCell {
...
+        }
+
+        let image = "schemes/icon-\(schemeIconSuffix)".image(forClass: CardListCell.self)
         schemeImageView.image = image
     }
 
+    func schemeIconSuffix(scheme: Card.Scheme) -> String? {
+        switch scheme {
+        case .amex:
+            return "amex"
```
Comment (deepesh-vasthimal-cko) [inline Source/UI/CardListCell.swift:94]: Can move strings like  "amex" etc into one constants place to be used elsewhere.

File: Source/CardUtils.swift [hunk 7]
```diff
@@ -1,65 +1,19 @@
 import Foundation
+import Checkout
 
 /// Payment Form Utilities
 public class CardUtils {
```
Comment (deepesh-vasthimal-cko) [inline Source/CardUtils.swift:5]: Do we need this ? I am expecting this should be inside checkout iOS SDK

**Review by deepesh-vasthimal-cko — COMMENTED** [review]
Few minor comments please have a look otherwise looks good.

File: Source/CheckoutSdkIos.swift (hunk 4 above)
Comment (harry-brown-cko) [reply Source/CheckoutSdkIos.swift:7]: This is a raw function that just lives in the sdk (its not part of any struct or class or whatever). its used to add the toolbar with the "done" button above the software keyboard when typing. Only change I've made here is to no longer make it public.

When Frames' UI is rewritten this whole file should be deleted, but for now this is still required.

File: Source/CardUtils.swift (hunk 7 above)
Comment (harry-brown-cko) [reply Source/CardUtils.swift:5]: card gaps and formatting are currently not part of the checkout sdk. we can move them over in future tickets

File: Source/UI/Views/CvvInputView.swift [hunk 8]
```diff
@@ -44,10 +54,23 @@ import UIKit
...
+        }
+
+        switch cardValidator.validate(cvv: cvv, cardScheme: scheme) {
+        case .success:
+            return true
+        case .failure(let error):
+            switch error {
+            case .containsNonDigits:
+                return false
+            // TODO: we should add an error case for incomplete CVV
```
[the remark this answers is not in the document]
Comment (harry-brown-cko) [reply Source/UI/Views/CvvInputView.swift:69]: https://checkout.atlassian.net/browse/PIMOB-958

File: Source/UI/Views/CardNumberInputView.swift [hunk 9]
```diff
@@ -78,36 +78,32 @@ import Checkout
...
-                let lastChar = cardNumberFormatted[cardNumberFormatted.index(cardNumberFormatted.startIndex,
-                                                                             offsetBy: targetCursorPosition - 1)]
-                if lastChar == " " && previousTextCount < cardNumberFormatted.count {
-                    targetPosition = textField
-                        .position(from: textField.beginningOfDocument, offset: targetCursorPosition + 1)!
-                }
-            }
-            textField.selectedTextRange = textField.textRange(from: targetPosition, to: targetPosition)
-        }
+    // TODO: I think we should delete this block. Unsure what it does
```
[the remark this answers is not in the document]
Comment (harry-brown-cko) [reply Source/UI/Views/CardNumberInputView.swift:95]: its something awkward to do with removing a space at the end of the string or something? If we need this logic it should be rewritten anyway

File: Source/UI/Views/CardNumberInputView.swift [hunk 10]
```diff
@@ -78,36 +78,32 @@ import Checkout
...
-        //            return
-        //        }
-
-        let cardNumber = cardsUtils!.standardize(cardNumber: textField.text!)
-        let cardType = cardsUtils.getTypeOf(cardNumber: cardNumber)
-        guard let cardTypeUnwrap = cardType else { return }
-        delegate?.onChangeCardNumber(cardType: cardType)
+        guard let cardUtils = cardUtils,
+              let rawText = textField.text else {
+            return
```
[the remark this answers is not in the document]
Comment (harry-brown-cko) [reply Source/UI/Views/CardNumberInputView.swift:83]: Same behaviour as previous. Will be fixed as part of UI redesign anyway

File: Source/UI/Controllers/CardViewController.swift [hunk 11]
```diff
@@ -377,23 +372,26 @@ public class CardViewController: UIViewController,
...
+    public func onChangeCardNumber(scheme: Card.Scheme) {
         // reset if the card number is empty
-        if cardType == nil && lastSelected != nil {
+        if scheme == .unknown && lastSelected != nil {
             cardView.schemeIconsStackView.arrangedSubviews.forEach { $0.alpha = 1 }
             lastSelected = nil
         }
-        guard let type = cardType else { return }
-        let index = availableSchemes.firstIndex(of: type.scheme)
+        guard scheme != .unknown else {
```
[the remark this answers is not in the document]
Comment (harry-brown-cko) [reply Source/UI/Controllers/CardViewController.swift:390]: Unknown scheme is returned whenever an invalid card number is entered. A validation error will be shown to the user from elsewhere

File: Source/CardUtils.swift (hunk 7 above)
Comment (deepesh-vasthimal-cko) [reply Source/CardUtils.swift:5]: Task : https://checkout.atlassian.net/browse/PIMOB-1193

File: Source/CardUtils.swift (hunk 1 above)
Comment (deepesh-vasthimal-cko) [reply Source/CardUtils.swift:9]: Task : https://checkout.atlassian.net/browse/PIMOB-1193

File: Source/CardUtils.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (deepesh-vasthimal-cko) [reply Source/CardUtils.swift:35]: Task : https://checkout.atlassian.net/browse/PIMOB-1193

**Review by deepesh-vasthimal-cko — APPROVED** [review]

File: Source/CardUtils.swift (hunk 7 above)
Comment (ehab-al-cko) [inline Source/CardUtils.swift:5]:
```suggestion
public final class CardUtils {
```

File: Source/CheckoutSdkIos.swift (hunk 4 above)
Comment (ehab-al-cko) [inline Source/CheckoutSdkIos.swift:7]:
```suggestion
private func addKeyboardToolbarNavigation(textFields: [UITextField]) {
```

File: Source/Core/Constants.swift [hunk 12]
```diff
@@ -1,7 +1,18 @@
-struct Constants {
-    
+enum Constants {
```
Comment (ehab-al-cko) [inline Source/Core/Constants.swift:1]:
```suggestion
@frozen enum Constants {
```

File: Source/Core/Environment.swift [hunk 13]
```diff
@@ -1,40 +1,14 @@
...
-    var unifiedPaymentsURL: URL { get }
-    
-}
-
 /// Checkout API Environment
 ///
 /// - live
 /// - sandbox
-public enum Environment: String, EnvironmentURLProviding {
+public enum Environment: String {
```
Comment (ehab-al-cko) [inline Source/Core/Environment.swift:7]:
```suggestion
public @frozen enum Environment: String {
```

File: Source/UI/Controllers/CardViewController.swift [hunk 14]
```diff
@@ -377,23 +372,26 @@ public class CardViewController: UIViewController,
 
         if let superView = view as? CardNumberInputView {
             let cardNumber = superView.textField.text!
```
Comment (ehab-al-cko) [inline Source/UI/Controllers/CardViewController.swift:374]:
```suggestion
        if let superView = view as? CardNumberInputView , 
           let cardNumber = superView.textField.text {
```

**Review by ehab-al-cko — APPROVED** [review]

File: Source/CheckoutSdkIos.swift (hunk 4 above)
Comment (harry-brown-cko) [reply Source/CheckoutSdkIos.swift:7]: This function is used across multiple views in Frames. We should keep it non-private for now

**Pull request merged**